contracts/0.8.9/WithdrawalQueue.sol

```diff
@@ -148,64 +149,94 @@ contract WithdrawalQueue {
-            revert LidoDAOAgentZeroAddress();
+    /**
+     * @notice Intialize the contract storage explicitly. NB! It's initialized in paused state by default 
+     * @param _admin admin address that can change every role.
+     * @param _pauser address that will be able to pause the withdrawals
+     * @param _resumer address that will be able to resume the withdrawals after pause
+     * @param _finalizer address that can finalize requests in the queue
+     * @dev Reverts with `AdminZeroAddress()` if `_admin` equals to `address(0)`
+     */
+    function initialize(address _admin, address _pauser, address _resumer, address _finalizer) external {
+        if (_admin == address(0)) {
+            revert AdminZeroAddress();
         }
 
-        _initialize(_lidoDAOAgent);
+        _initialize(_admin, _pauser, _resumer, _finalizer);
     }
 
-    /// @notice Resume new withdrawal requests placement
-    function resumeRequestsPlacement() external whenInitialized whenPaused onlyLidoDAOAgent {
-        REQUESTS_PLACEMENT_RESUMED_POSITION.setStorageBool(true);
-
-        emit WithdrawalRequestsPlacementResumed();
+    /// @notice Returns whether the contract is initialized or not
+    function isInitialized() external view returns (bool) {
+        return CONTRACT_VERSION_POSITION.getStorageUint256() != 0;
     }
 
-    /// @notice Pause new withdrawal requests placement
-    function pauseRequestsPlacement() external whenResumed onlyLidoDAOAgent {
-        REQUESTS_PLACEMENT_RESUMED_POSITION.setStorageBool(false);
+    /**
+     * @notice Resume withdrawal requests placement and finalization
+     * @dev Reverts with `Uninitialized()` if contract is not initialized
+     * @dev Reverts with `PausedExpected()` if contract is already resumed
+     * @dev Reverts with `AccessControl:...` reason if sender has no `RESUME_ROLE`
+     */
+    function resume() external whenInitialized whenPaused onlyRole(RESUME_ROLE) {
+        RESUMED_POSITION.setStorageBool(true);
 
-        emit WithdrawalRequestsPlacementPaused();
+        emit WithdrawalQueueResumed();
     }
 
     /**
-     * @notice Getter for withdrawal queue length
-     * @return length of the request queue
+     * @notice Pause withdrawal requests placement and finalization. Claiming finalized requests will still be available
+     * @dev Reverts with `ResumedExpected()` if contract is already paused
+     * @dev Reverts with `AccessControl:...` reason if sender has no `PAUSE_ROLE`
      */
+    function pause() external whenResumed onlyRole(PAUSE_ROLE) {
+        RESUMED_POSITION.setStorageBool(false);
+
+        emit WithdrawalQueuePaused();
+    }
+
+    /// @notice Returns whether the requests placement and finalization is paused or not
+    function isPaused() external view returns (bool) {
+        return !RESUMED_POSITION.getStorageBool();
+    }
+
+    /// @notice Returns the length of the withdrawal request queue 
     function queueLength() external view returns (uint256) {
```

Comment:
maybe add the helpers `unfinalizedQueueLength` and `unfinalizedStETH` also for the frontend devs?